StartEycaActivationOrchestrator/index.ts

```diff
@@ -71,12 +78,26 @@ export const handler = function*(
       trackExAndThrow(e, "eyca.activate.exception.decode.activityOutput")
     );
 
-    if (updateEycaResult.kind !== "SUCCESS") {
-      trackExAndThrow(
-        new Error("Cannot activate EYCA Card"),
-        "eyca.activate.exception.failure.activityOutput"
+    const storedEycaCard = ActivityResultSuccessWithValue.decode(
+      updateEycaResult
+    )
+      .map(val => val.value)
+      .getOrElseL(() =>
+        trackExAndThrow(
+          new Error("Cannot activate EYCA Card"),
+          "eyca.activate.exception.failure.activityOutput"
+        )
```

Comment:
I want to raise an error if the values is not what I expect it to be. 
From what I can see, `fold` ask me to return an element of the same type of the right path

Reply:
> I want to raise an error if the value is not what I expect it to be.
> From what I can see, `fold` ask me to return an element of the same type of the right path

Not necessary :), you can try to add a test to verify this behaviour 💪

Reply:
Can you show me an example somewhere in code?

Reply:
:) https://github.com/pagopa/io-functions-cgn/blob/bbd665a236a871838dfb5d74a15c382cb6f9958c/Info/handler.ts#L24

Reply:
Ok, in that code you are merging together two different objects and returning the same object type in each case.
In this particoular case I need to retrieve the `right` data and raise an exception otherwise.
I tried with `map` and `mapLeft`, but since we are within an Orchestrator it doesn't build, because of the `yield` reserved keyword

```suggestion
        ActivityResultSuccessWithValue.decode(updateEycaResult)
      .map(val => val.value)
      .map(
        activatedEycaCard =>
          yield context.df.callActivityWithRetry(
            "StoreEycaExpirationActivity",
            internalRetryOptions,
            StoreEycaExpirationActivityInput.encode({
              activationDate: activatedEycaCard.activation_date,
              expirationDate: activatedEycaCard.expiration_date,
              fiscalCode
            })
          )
      )
      .mapLeft(() =>
        trackExAndThrow(
          new Error("Cannot activate EYCA Card"),
          "eyca.activate.exception.failure.activityOutput"
        )
      );
```

Reply:
I mean something like this:
```
const storedEycaCard = ActivityResultSuccessWithValue.decode(
      updateEycaResult
    ).fold(
      () =>
        trackExAndThrow(
          new Error("Cannot activate EYCA Card"),
          "eyca.activate.exception.failure.activityOutput"
        ),
      result => result.value
    );

yield context.df.callActivityWithRetry(
      "StoreEycaExpirationActivity",
      internalRetryOptions,
      StoreEycaExpirationActivityInput.encode({
        activationDate: storedEycaCard.activation_date,
        expirationDate: storedEycaCard.expiration_date,
        fiscalCode
      })
    );
```

Reply:
Ok, now I got it, thank you!! Much better now, I wasn't considering the `never` return type..
Done in [#77ce530](https://github.com/pagopa/io-functions-cgn/pull/27/commits/77ce530144b747f250b0940700821ff946c1107c)